seisbench/data/base.py

```diff
@@ -442,7 +442,7 @@ def _get_component_mapping(self, source, target):
         :param target:
         :return:
         """
-        source = list(source)
+        source = list(str(source))
```

Comment:
I'm not sure about this change. Why is it required? I think the other change should already encompass this option. It also breaks the possibility to use a list as source.

Reply:
You're actually right. But pandas treats empty strings as NaN and therefore as floats. I added a type check that raises a ValueError in such a case. But it could also be taken care of directly after reading the metadata file, or even not at all since it is rather an error of pre-processing. I'd leave this decision to you.